streamer/src/nonblocking/quic.rs

```diff
@@ -236,6 +237,51 @@ pub fn spawn_server_multi(
     })
 }
 
+/// struct ease tracking connections of all stages, so that we do not have to
+/// litter the code with open connection tracking. This is added into the
+/// connection table as part of the ConnectionEntry. The reference is auto
+/// reduced when it is dropped.
+
+struct ClientConnectionTracker {
+    stats: Arc<StreamerStats>,
+    prev_open_connections: usize,
+}
+
+/// This is required by ConnectionEntry for supporting debug format.
+impl fmt::Debug for ClientConnectionTracker {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("StreamerClientConnection")
+            .field(
+                "open_connections:",
+                &self.stats.open_connections.load(Ordering::Relaxed),
+            )
+            .finish()
+    }
+}
+
+impl Drop for ClientConnectionTracker {
+    /// When this is dropped, reduce the open connection count.
+    fn drop(&mut self) {
+        self.stats.open_connections.fetch_sub(1, Ordering::Relaxed);
+    }
+}
+
+impl ClientConnectionTracker {
+    /// Create StreamerClientConnection and increment open connection count.
+    fn new(stats: Arc<StreamerStats>) -> Self {
+        let prev_open_connections = stats.open_connections.fetch_add(1, Ordering::Relaxed);
+        Self {
+            stats,
+            prev_open_connections,
+        }
+    }
+
+    /// Get the previous open connections before our increment.
+    fn get_prev_open_connections(&self) -> usize {
```

Comment:
This API looks strange to me, instead of doing this, I'd have
`ClientConnectionTracker::new(stats: Arc<StreamStats>, max: usize) ->
Result<Self, ()>` and return Err if there are too many connections

Reply:
Good idea. Done